src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -59,4 +59,182 @@ private bool IsMainWindow(IntPtr handle)
             return true;
         }
     }
+
+    internal static partial class NtProcessManager
+    {
+        private static ProcessModuleCollection GetModules(int processId, bool firstModuleOnly)
+        {
+            // preserving Everett behavior.
+            if (processId == SystemProcessID || processId == IdleProcessID)
+            {
+                // system process and idle process doesn't have any modules
+                throw new Win32Exception(Interop.Errors.EFail, SR.EnumProcessModuleFailed);
+            }
+
+            SafeProcessHandle processHandle = SafeProcessHandle.InvalidHandle;
+            try
+            {
+                processHandle = ProcessManager.OpenProcess(processId, Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION | Interop.Advapi32.ProcessOptions.PROCESS_VM_READ, true);
+
+                IntPtr[] moduleHandles = new IntPtr[64];
+                GCHandle moduleHandlesArrayHandle = new GCHandle();
+                int moduleCount = 0;
+                for (;;)
+                {
+                    bool enumResult = false;
+                    try
+                    {
+                        moduleHandlesArrayHandle = GCHandle.Alloc(moduleHandles, GCHandleType.Pinned);
```

Comment:
`GCHandle.Alloc` is unnecessary here. `fixed` is more efficient (smaller, faster code).